libs/python/open_model_zoo.py

```diff
@@ -41,71 +41,30 @@ def __init__(self, model_name, *, precisions=None, download_dir='models', cache_
-        (if the download specified only one precision value) or throws an error 
-        if the download specified few precision values or downloading precision value is None.
-        If there is already converted model or the model has been downloaded in
-        IR - just returns the paths to the files.
-
-        Parameters
-        ----------
-            precision
-                Target model precision. By default is None.
-
-        Returns
-        -------
-            xml_path
-                Path to generated or downloaded .xml file
-            bin_path
-                Path to generated or downloaded .bin file
-        '''
-        if self.downloaded_precisions:
-            if len(self.downloaded_precisions) > 1 and (not precision or precision not in self.downloaded_precisions):
-                raise ValueError('Incorrect precision value!\n    Current precision value is: \''
-                                            + str(precision) +  '\'\n    Downloaded model\'s precision values are: ' 
-                                            + str(self.downloaded_precisions))
-            
-            if len(self.downloaded_precisions) == 1 and precision and precision != self.downloaded_precisions[0]:
-                raise ValueError('Downloaded ({}) and requested ({}) precisions do not match'.format(self.downloaded_precisions[0], precision))
-
-        res_precision = precision if precision else self.downloaded_precisions[0]
-        if res_precision not in self.precisions:
+        if precision not in topology.precisions:
             raise ValueError('Incorrect precision value!\n    Current precision value is: \'' 
-                                        + str(res_precision) + '\'.\n    Allowed precision values are: ' 
-                                        + str(self.precisions))
+                                        + str(res_precision) + '\'.\n    Allowed precision values for model are: ' 
+                                        + str(topology.precisions))
   
-        prefix = Path(self.model_dir) / res_precision / self.name
+        prefix = Path(model_dir) / precision / model_name
 
-        xml_path = str(prefix) + '.xml'
-        bin_path = str(prefix) + '.bin'
-        if os.path.exists(xml_path) and os.path.exists(bin_path):
-            return xml_path, bin_path
+        self.xml_path = str(prefix) + '.xml'
+        self.bin_path = str(prefix) + '.bin'
 
-        converter.main(['--name=' + self.name, '--precisions=' + res_precision,
-                        '--download_dir=' + str(self.download_dir)])
+        if not os.path.exists(self.xml_path) or not os.path.exists(self.bin_path):
+            converter.main(['--name=' + model_name, '--precisions=' + res_precision,
+                        '--download_dir=' + str(download_dir)])
 
-        return xml_path, bin_path
+    def description(self):
+        return self._description
+
+    def task_type(self):
+        return self._task_type
```

Comment:
I think that now it's OK to keep them as fields (just `description` and `task_type`)